### ISSUE TITLE
when use \Doctrine\DBAL\Schema\Comparator to compare two table schema, has a problem

### ISSUE THREAD

**Issue by bryantyan** [doctrine/dbal#2302]
when use \Doctrine\DBAL\Schema\Comparator to compare two table schema.

testing 2.5.4 on mysql

from table schema:
```
CREATE TABLE `cart` (
  `cart_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `user_id` int(11) NOT NULL DEFAULT '-1' ,
  PRIMARY KEY (`cart_id`,`user_id`),
) ENGINE=InnoDB CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

to table schema:
```
CREATE TABLE `cart` (
  `cart_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `user_id` int(11) NOT NULL DEFAULT '-1' ,
  PRIMARY KEY (`cart_id`),
) ENGINE=InnoDB AUTO_INCREMENT=222 DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci
```

And the schema diff sql is :

```
ALTER TABLE systrade_cart DROP PRIMARY KEY
ALTER TABLE systrade_cart ADD PRIMARY KEY (cart_id)
```

the execution wil fail: 

> **it say**:exception 'Doctrine\DBAL\Driver\Mysqli\MysqliException' with message 'Incorrect table definition; there can be only one auto column and it must be defined as a key' in vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php:150.

i find the code and old issue, guess this is a bug.

the bug like DBAL-464, but is difference

**Comment by deeky666** [comment]
@bryantyan patch created: #2303

**Comment by bryantyan** [comment]
another similar bug:

testing 2.5.4 on mysql

from table schema:
```
CREATE TABLE `cart` (
  `cart_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `user_id` int(11) NOT NULL DEFAULT '-1' ,
  PRIMARY KEY (`cart_id`),
) ENGINE=InnoDB CHARSET=utf8 COLLATE=utf8_unicode_ci
```

to table schema:

```
CREATE TABLE `cart` (
  `cart_id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `user_id` int(11) NOT NULL DEFAULT '-1' ,
  PRIMARY KEY (`cart_id`,`user_id`),
) ENGINE=InnoDB CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

And the schema diff sql is :

```
ALTER TABLE systrade_cart DROP PRIMARY KEY
ALTER TABLE systrade_cart CHANGE cart_id cart_id INT UNSIGNED AUTO_INCREMENT NOT NULL
ALTER TABLE systrade_cart ADD PRIMARY KEY (cart_id, user_id)
```

when exec `ALTER TABLE systrade_cart CHANGE cart_id cart_id INT UNSIGNED AUTO_INCREMENT NOT NULL`

the execution will fail:

>**it say**: exception 'Doctrine\DBAL\Driver\Mysqli\MysqliException' with message 'Incorrect table definition; there can be only one auto column and it must be defined as a key' in /Users/bryant/codes/bbc/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php:150

**Comment by deeky666** [comment]
@bryantyan this is also fixed by #2303. But to verify and avoid regressions I added another test case for this here https://github.com/deeky666/dbal/commit/33cfdce9712c40a2d7e8fa681de494a4afa3a7db

**Comment by Ocramius** [comment]
Handled in #2303
